test/tests/functional/pbs_daemon_service_user.py

```diff
@@ -0,0 +1,130 @@
+
+import resource
+
+from tests.functional import *
+
+
+class TestDaemonServiceUser(TestFunctional):
+
+    """
+    Test suite to test running schedulers as a non-root user
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        self.orig_pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+
+    def tearDown(self):
+        self.du.set_pbs_config(self.server.shortname,
+                               confs=self.orig_pbs_conf,
+                               append=False)
+        TestFunctional.tearDown(self)
+
+    def common_test(self, binary, runas, scheduser, msg):
+        """
+        Test if running `binary` as `runas` with
+        PBS_DAEMON_SERVICE_USER set as `scheduser`
+        Check to see `msg` is in stderr
+        """
+        if scheduser:
+            self.du.set_pbs_config(
+                self.server.hostname,
+                confs={'PBS_DAEMON_SERVICE_USER': str(scheduser)}
+            )
+        else:
+            self.du.unset_pbs_config(
+                self.server.hostname,
+                confs='PBS_DAEMON_SERVICE_USER'
+            )
+        pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        binpath = os.path.join(pbs_conf['PBS_EXEC'], 'sbin', binary)
+        ret = self.du.run_cmd(self.server.shortname,
+                              cmd=[binpath], runas=runas)
+        self.assertEquals(ret['rc'], 1)
+        self.assertIn(msg, '\n'.join(ret['err']))
+
+    def test_sched_runas_root(self):
+        """
+        Test if running sched as root with
+        PBS_DAEMON_SERVICE_USER set as another user
+        """
+        self.common_test('pbs_sched', ROOT_USER, TEST_USER,
+                         'Must be run by PBS_DAEMON_SERVICE_USER')
+
+    def test_pbsfs_runas_root(self):
+        """
+        Test if running pbsfs as root with
+        PBS_DAEMON_SERVICE_USER set as another user
+        """
+        self.common_test('pbsfs', ROOT_USER, TEST_USER,
+                         'Must be run by PBS_DAEMON_SERVICE_USER')
+
+    def test_sched_runas_nonroot(self):
+        """
+        Test if running sched as nonroot with
+        PBS_DAEMON_SERVICE_USER set as another user
+        """
+        self.common_test('pbs_sched', TEST_USER, TEST_USER1,
+                         'Must be run by PBS_DAEMON_SERVICE_USER')
+
+    def test_pbsfs_runas_nonroot(self):
+        """
+        Test if running pbsfs as root with
+        PBS_DAEMON_SERVICE_USER set as another user
+        """
+        self.common_test('pbsfs', TEST_USER, TEST_USER1,
+                         'Must be run by PBS_DAEMON_SERVICE_USER')
+
+    def test_sched_runas_nonroot_notset(self):
+        """
+        Test if running sched as nonroot with
+        PBS_DAEMON_SERVICE_USER not set
+        """
+        self.common_test('pbs_sched', TEST_USER, None,
+                         'Must be run by PBS_DAEMON_SERVICE_USER if '
+                         'set or root if not set')
```

Comment:
I don't get this message fully.  Don't we know if PBS_DAEMON_SERVICE_USER is set?  Why is this message different than the other message?

Reply:
that's the log message that we're matching with. to give more context, i used the whole message to include it not being set.

Reply:
I understand you are matching to this message, that still doesn't explain the message.  Maybe I should have given this comment up when you spit out the message.